### PR TITLE
Fix for binance chains

### PR DESCRIPTION
**Pull request by Sahilgat**
Support some other small changes for Binance blockchain

### CODE REVIEW

File: ribbon/ribbon/swap.py [hunk 1]
```diff
@@ -165,12 +166,12 @@ def create_offer(self, offer: Offer, wallet: Wallet) -> str:
         offer.biddingToken = get_address(offer.biddingToken)
 
         nonce = self.w3.eth.get_transaction_count(wallet.public_key)
-        tx = self.contract.functions.createOffer(*list(asdict(offer).values())).build_transaction(
-            {
-                "nonce": nonce,
-                "gas": GAS_LIMIT,
-            }
-        )
+        if self.config.chain_id == Chains.BSC or self.config.chain_id == Chains.BSC_TESTNET:
```
Comment (pdonorio) [inline ribbon/ribbon/swap.py:169]:
```suggestion
        if self.config.chain_id in [Chains.BSC, Chains.BSC_TESTNET]:
```

File: ribbon/ribbon/contract.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def __init__(self, config: ContractConfig):
                 + f"({chain.value})"
             )
 
-        if chain == Chains.FUJI:
+        if chain == Chains.FUJI or chain == Chains.BSC or chain == Chains.BSC_TESTNET:
```
[the remark this answers is not in the document]
Comment (Sahilgat) [reply ribbon/ribbon/contract.py:63]: We need this when the consensus mechanism for the chain is proof-of-authority. More information here: https://stackoverflow.com/questions/70812529/the-field-extradata-is-97-bytes-but-should-be-32-it-is-quite-likely-that-you-a

File: ribbon/ribbon/swap.py (hunk 1 above)
Comment (mdantonio) [reply ribbon/ribbon/swap.py:169]: We can do that, but considering the other check (FUJI & BSC & BSC_TESTNET) I'm not sure it solves so much

Here would be better:

```
- if self.config.chain_id in [Chains.BSC, Chains.BSC_TESTNET]:
+ if self.config.chain_id.is_binance:
```

But in the other case would be something like:

```
- if chain in [Chains.FUJI, Chains.BSC, Chains.BSC_TESTNET]:
+ if chain.is_binance or (chain.is_avalanche and chain.is_testnet):
```

Comment (pdonorio) [reply ribbon/ribbon/swap.py:169]: Agreed. So my two suggestions should be enough, thank you 🙏🏽

**Comment by mdantonio** [comment]
We have some failures on both linters and mypy 😢

**Comment by pdonorio** [comment]
@mdantonio @Sahilgat fixed the linters, if you can 👁️

File: ribbon/ribbon/contract.py (hunk 2 above)
Comment (mdantonio) [inline ribbon/ribbon/contract.py:63]: We do not want to convert the ORs into `chain in [Chains.FUJI, Chains.BSC, Chains.BSC_TESTNET]`?

**Review by mdantonio — APPROVED** [review]

File: ribbon/ribbon/contract.py (hunk 2 above)
Comment (pdonorio) [reply ribbon/ribbon/contract.py]: Done

**Review by pdonorio — APPROVED** [review]

**Pull request merged**